src/components/Header/Header.tsx

```diff
@@ -0,0 +1,56 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef } from 'react';
+
+type Props = {
+  handleSubmit: (e: React.FormEvent) => void,
+  handleTitleChange: (value: string) => void,
+  handleToggleAll: () => void,
+  todoTitle: string,
+  isAdding: boolean,
+  toggleAll: boolean,
+};
+
+export const Header: React.FC<Props> = React.memo(
```

Comment:
There is no sense in using memo. for example handleSubmit is not wrapped by useCallback so it is new function on each App render, so props change on each render, so memo never works